File: src/main/java/net/finance/utils/EncryptUtils.java
package net.finance.utils;

import java.nio.charset.Charset;
import java.security.MessageDigest;
import java.util.Optional;

import org.springframework.security.crypto.codec.Hex;

public class EncryptUtils {

	public static Optional<String> hashPassword(final String password) {
		try {
			final MessageDigest dig = MessageDigest.getInstance("MD5");
			dig.reset();
			dig.update(password.getBytes(Charset.forName("UTF8")));
			final byte[] arrBytes = dig.digest();
			return Optional.of(new String(Hex.encode(arrBytes)));
		} catch (final Exception e) {
			return Optional.empty();
		}
	}
}

File: src/main/kotlin/com/finances/utils/EncryptUtils.kt
package com.finances.utils

import org.springframework.security.crypto.codec.Hex
import java.nio.charset.Charset
import java.security.MessageDigest
import java.util.*

object EncryptUtils {

    @JvmStatic
	fun hashPassword(password: String): Optional<String> {
        return try {
            val dig = MessageDigest.getInstance("MD5")
            dig.reset()
            dig.update(password.toByteArray(Charset.forName("UTF8")))
            val arrBytes = dig.digest()
            Optional.of(String(Hex.encode(arrBytes)))
        } catch (e: Exception) {
            Optional.empty()
        }
    }

    @JvmStatic
    fun main(args: Array<String>) {
        println(hashPassword("admin").get())
    }
}